stl/inc/algorithm

```diff
@@ -2841,6 +2841,85 @@ _FwdIt2 replace_copy_if(_ExPo&&, _FwdIt1 _First, _FwdIt1 _Last, _FwdIt2 _Dest, _
 }
 #endif // _HAS_CXX17
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // VARIABLE ranges::fill
+    class _Fill_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        template <class _Ty, output_iterator<const _Ty&> _It, sentinel_for<_It> _Se>
+        constexpr _It operator()(_It _First, _Se _Last, const _Ty& _Value) const {
+            _Adl_verify_range(_First, _Last);
+            auto _UFirst      = _Get_unwrapped(_STD move(_First));
+            const auto _ULast = _Get_unwrapped(_STD move(_Last));
+            if constexpr (_Fill_memset_is_safe<decltype(_UFirst), _Ty>) {
+#ifdef __cpp_lib_is_constant_evaluated
+                if (!_STD is_constant_evaluated())
+#endif // __cpp_lib_is_constant_evaluated
+                {
+                    const auto distance = static_cast<size_t>(_ULast - _UFirst);

```

Comment:
`distance` should be `_Ugly`.

(Technically, as the name of a Standard function, it's immune to macroization. However, we conventionally avoid reusing/shadowing Standard names; this makes searching easier, and the convention "everything must be `_Ugly`" is easier to follow. The rare exceptions to this rule are `type`, `value`, and occasionally `pointer`.)